modules/services/colima.nix

```
@@ -0,0 +1,267 @@
+    };
+    perlPackage = lib.mkPackageOption pkgs "perl" {
+      extraDescription = "Used by colima during image download for the shasum command.";
+    };
+    sshPackage = lib.mkPackageOption pkgs "openssh" {
+      extraDescription = "Used by colima to manage the vm.";
+    };
+    coreutilsPackage = lib.mkPackageOption pkgs "coreutils" {
+      extraDescription = "Used in various ways by colima.";
+    };
+    curlPackage = lib.mkPackageOption pkgs "curl" {
+      extraDescription = "Used by colima to donwload images.";
+    };
+    bashPackage = lib.mkPackageOption pkgs "bashNonInteractive" {
+      extraDescription = "Used by colima's internal scripts.";
+    };
+
+    profiles = lib.mkOption {
+      default = {
+        default = {
+          isActive = true;
+          isService = true;
+        };
+      };
+      description = ''
+        Profiles allow multiple colima configurations. The default profile is active by default.
+        If you have used colima before, you may need to delete existing configuration using `colima delete` or use a different profile.
+
+        Note that removing a configured profile will not delete the corresponding Colima instance.
+        You will need to manually run `colima delete <profile-name>` to remove the instance and release resources.
+      '';
+      example = ''
+        {
+          default = {
+            isActive = true;
+            isService = true;
+          };
+          rosetta = {
+            isService = true;
+            settings.rosetta = true;
+          };
+          powerful = {
+            settings.cpu = 8;
+          };
+        };
+      '';
+      type = lib.types.attrsOf (
+        lib.types.submodule (
+          { name, ... }:
+          {
+            options = {
+              name = lib.mkOption {
+                type = lib.types.str;
+                default = name;
+                readOnly = true;
+                description = "The profile's name.";
+              };
+
+              isService = lib.mkOption {
+                type = lib.types.bool;
+                default = false;
+                example = true;
+                description = ''
+                  Whether this profile will run as a service.
+                '';
+              };
+
+              isActive = lib.mkOption {
+                type = lib.types.bool;
+                default = false;
+                example = true;
+                description = ''
+                  Whether to set this profile as:
+                  - active docker context
+                  - active kubernetes context
+                  - active incus remote
+                  Exactly one or zero profiles should have this option set.
+                '';
+              };
+
+              logFile = lib.mkOption {
+                type = lib.types.path;
+                default = "${config.home.homeDirectory}/.local/state/colima-${name}.log";
```

```suggestion
                default = ".local/state/colima-${name}.log";
```

`home.file` would already use this path

---

Hmm I'm not quite sure what you mean. I don't think I can use home.file here?